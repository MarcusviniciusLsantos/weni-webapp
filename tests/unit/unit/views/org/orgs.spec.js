import { vi } from 'vitest';
import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';
import Projects from '@/views/org/orgs.vue';
import OrgList from '@/components/orgs/orgList.vue';
import { org } from '../../../__mocks__';

vi.mock('@/api/request.js', () => {});

describe('orgs.vue', () => {
  let wrapper;
  let state;
  let store;
  let actions;

  beforeEach(() => {
    const mutations = {
      setOrgStatus(state, status) {
        state.Org.orgs.status = status;
      },
    };

    state = {
      Org: {
        orgs: { data: [org], status: 'complete' },
      },
    };

    actions = {
      clearCurrentOrg: vi.fn(),
      clearCurrentProject: vi.fn(),
    };

    store = createStore({
      state,
      mutations,
      actions,
    });

    wrapper = shallowMount(Projects, {
      global: {
        plugins: [store],
        stubs: {
          RouterLink: RouterLinkStub,
          OrgList,
          SkeletonLoading: true,
          UnnnicButton: true,
          UnnnicIconSvg: true,
          UnnnicInput: true,
          UnnnicSkeletonLoading: true,
        },
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  // SHOULD WE TEST IT AFTER.
  // describe('tryAgain()', () => {
  //   it('should call the function tryAgain', () => {
  //     const spy = vi.spyOn(wrapper.vm.$refs.orgList, 'reloadOrganizations');
  //     expect(spy).not.toHaveBeenCalled();
  //     wrapper.vm.tryAgain();
  //     expect(spy).toHaveBeenCalledTimes(1);
  //   });
  // });

  describe('organizationsStatus() watcher', () => {
    it('Should set error true', async () => {
      store.commit('setOrgStatus', 'error');
      await wrapper.vm.$nextTick();
      expect(wrapper.vm.error).toBe(true);
    });

    it('Should test loaded status for error', async () => {
      store.commit('setOrgStatus', 'loaded');
      await wrapper.vm.$nextTick();
      expect(wrapper.vm.error).toBe(false);
    });

    it('Should test empty status for error', async () => {
      store.commit('setOrgStatus', 'empty');
      await wrapper.vm.$nextTick();
      expect(wrapper.vm.error).toBe(false);
    });
  });
});
