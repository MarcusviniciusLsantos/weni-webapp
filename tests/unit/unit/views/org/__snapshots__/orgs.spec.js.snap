// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`orgs.vue > should be rendered properly 1`] = `
<div
  class="weni-orgs status-complete"
  data-v-04c76209=""
>
  <div
    data-v-04c76209=""
  >
    <div
      class="weni-orgs__left"
      data-v-04c76209=""
    >
      <div
        class="page-header"
        data-v-04c76209=""
      >
        <!--v-if-->
        <div
          data-v-04c76209=""
        >
          <h1
            data-v-04c76209=""
          >
            Your organizations
          </h1>
          <p
            data-v-04c76209=""
          >
            Listed below are the organizations you belong to. Choose or create a new one to access the platform.
          </p>
        </div>
        <a
          data-v-04c76209=""
        >
          <unnnic-button-stub
            class="create-org-button"
            data-v-04c76209=""
            iconleft="add-1"
            type="primary"
          />
        </a>
      </div>
    </div>
    
    <hr
      data-v-04c76209=""
    />
    <div
      class="unnnic-grid-span-5 weni-orgs__right"
      data-v-04c76209=""
    >
      <div
        class="weni-orgs__list"
        data-v-04c76209=""
      >
        <div
          class="filters"
          data-v-04c76209=""
        >
          <unnnic-input-stub
            data-v-04c76209=""
            iconleft="search-1"
            modelvalue=""
            placeholder="Search"
            size="md"
          />
          <list-ordinator-stub
            data-v-04c76209=""
            ordinators="alphabetical,newer,older"
          />
        </div>
        <div
          class="weni-org-list__wrapper list-container"
          data-v-04c76209=""
          data-v-f419c7e3=""
        >
          <div
            class="weni-org-list"
            data-v-f419c7e3=""
          >
            
            <org-card-stub
              data-v-f419c7e3=""
              description="Description Lorem ipsum"
              name="Name Lorem ipsum"
              plan="trial"
            />
            
            <new-infinite-loading-stub
              complete="true"
              data-v-f419c7e3=""
              modelvalue="false"
              value="false"
            />
          </div>
        </div>
      </div>
    </div>
    
  </div>
</div>
`;
